rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractor.java

```diff
@@ -45,8 +46,14 @@ public CovidPatientHistoryActivityInteractor() {
                 }
                 String text = formWidgetKeyToTextMap.get(obs.getFormSubmissionField());
                 if (StringUtils.isNotBlank(text)) {
-                    PatientHistoryEntry patientHistoryEntry = new PatientHistoryEntry(text,
-                            getValues(obs, formWidgetKeyToTextMap));
+                    PatientHistoryEntry patientHistoryEntry;
+                    String value = getValues(obs, formWidgetKeyToTextMap);
+                    try {
+                        JSONArray jsonArray = new JSONArray(value);
+                        patientHistoryEntry = new PatientHistoryEntry(text, String.valueOf(jsonArray.length()));
+                    } catch (JSONException ex) {
+                        patientHistoryEntry = new PatientHistoryEntry(text, value);
+                    }
```

Comment:
I think we should figure out why https://github.com/onaio/rdt-standard/blob/20cc980aa0e0114acf36c6f83e8ef5da90c69294/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractor.java#L70 is failing. For repeating groups, only the "_count" object should be used as a value and not the original repeating group JSON.

Reply:
I have debugged the code and there is no field found which has a postfix `_count`. Here is the event data that is extracted from DB for example.
```json
{
      "fieldType": "formsubmissionField",
      "fieldDataType": "text",
      "fieldCode": "has_close_contact_with_covid_patient",
      "parentCode": "",
      "values": [
        "[{\"key\":\"name_label\",\"type\":\"label\",\"text\":\"Contacts name\",\"text_color\":\"#000000\"},{\"key\":\"name\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"edit_text\",\"v_required\":{\"value\":\"false\",\"err\":\"\"}},{\"key\":\"phone_number_lbl\",\"type\":\"label\",\"text\":\"Contacts phone number\",\"text_color\":\"#000000\"},{\"key\":\"phone_number\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"edit_text\",\"v_required\":{\"value\":\"false\",\"err\":\"\"},\"v_numeric\":{\"value\":\"true\",\"err\":\"Value must be numeric\"}},{\"key\":\"relationship_lbl\",\"type\":\"label\",\"text\":\"Relationship with contact\",\"text_color\":\"#000000\"},{\"key\":\"relationship\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"edit_text\",\"v_required\":{\"value\":\"false\",\"err\":\"\"}},{\"key\":\"date_of_first_contact\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"date_picker\",\"hint\":\"Date of first contact\",\"expanded\":false,\"duration\":{\"label\":\"Date of first contact\"},\"min_date\":\"today-100y\",\"max_date\":\"today\"},{\"key\":\"date_of_last_contact\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"date_picker\",\"hint\":\"Date of last contact\",\"expanded\":false,\"duration\":{\"label\":\"Date of last contact\"},\"min_date\":\"today-100y\",\"max_date\":\"today\"}]"
      ],
      "set": [],
      "formSubmissionField": "has_close_contact_with_covid_patient",
      "humanReadableValues": [],
      "saveObsAsArray": false
 }
```

Reply:
Ok, do we know why?

Reply:
As I pasted the above JSON there is not count field in the DB event.

Reply:
Could you find out why the count is not populated?

Reply:
I think because the event entity doesn't have the count field in the database.

Reply:
Why is that the case? Any time the repeating group is populated with a value, the count object is populated as well. Could you investigate cases in which this does not happen?

Reply:
yes, we creating a _count field with a value of reference_edit_text_hint if the field type is repeating_group
https://github.com/onaio/rdt-standard/blob/20cc980aa0e0114acf36c6f83e8ef5da90c69294/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/FormKeyTextExtractionUtil.java#L48-L51
 but the problem is that we don't have any count field in the `event`.
https://github.com/onaio/rdt-standard/blob/20cc980aa0e0114acf36c6f83e8ef5da90c69294/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractor.java#L42
this is the example obs which has no count field https://github.com/onaio/rdt-standard/pull/635#discussion_r532389271

Reply:
I understand that the count field is not present. I'm trying to understand in what instances that happens. Could you try playing around with the repeating group and check whether the count is populated in the Event? Are you able to replicate the case when the count is not recorded in the Event?

Reply:
OK, I will update you after creating another record and investigate then

Reply:
I have investigated the `addEvent` method in `EventClientRepo`. There is no operation found where the repeating_group field count has been processed in the event JSON.

Reply:
When you fill in data on a repeating group, does the count appear in the Event?

Reply:
no

Reply:
For what input? I've tried it myself and the numbers display correctly.